pkg/kncloudevents/message_sender_new.go

```diff
@@ -0,0 +1,182 @@
+/*
```

Comment:
why this "new" file instead of the same modified file?

Reply:
So that we can remove the whole file with the "old" code, after everything migrated. Simply to keep them separate.

Reply:
I'd just break the API tbh to avoid creating tech dept that we need to maintain

Reply:
I am not sure, if I understood you correctly. But the "old" methods are still used in the `HTTPMessageSender`. Until we migrated to use `CloudEventsRequest`, we have to keep them. WDYT about creating a seperate issue and adding it to the eventing TLS board, so we don't forget to clean it up?